### DISCUSSION TITLE
What should be used instead of deprecated CachedChatSubscription and others like this?

### DISCUSSION THREAD

**Discussion by Lao-Ax** [RocketChat/Rocket.Chat#32808] [Q&A] [answered]
What should be used instead of deprecated CachedChatSubscription and others like
- ChatSubscription
- RolesCollection
- etc

What is the proper substitute?
Or at least why they are deprecated?

**Comment by SouparnaChatterjee** [comment]
Hey! I have been exploring various features of Rocket.Chat. The collections like 'CachedChatSubscription', 'RolesCollection' etc. have been deprecated. This could be to achieve the following:

- Improve scalability and maintainability.
- Streamline the code base.
- Enhance performance
- Reduce technical debt

1. The 'Subscriptions' collection could be used to replace 'CachedChatSubscription' and 'ChatSubscription'. This collection acts as the foremost data source for all subscription related information in Rocket.Chat.
2. RolesCollection has been replaced by 'Roles' collection. This efficiently manages permissions and user roles. It implements the latest methods for role based access control.

Hope i have been helpful! :)

**Reply by Lao-Ax** [reply]
Thanks! I'll double check these classes.

**Reply by Lao-Ax** [reply]
Hi! It took a bit more time than I thought.

I can't find Roles collection. I see only apps/meteor/app/models/client/models/Roles.ts - it's deprecated.
The same for Subscriptions - there is no such a file on the client side. 

Could you be a bit more specific? Which files you were speaking about?

**Comment by tassoevan** [comment, marked as answer]
Addressed by https://github.com/RocketChat/Rocket.Chat/pull/33213

**Reply by Lao-Ax** [reply]
May I ask you to give me an example of existing code with Minimongo and Tanstack? Just a file name )

**Comment by Aurangzeb-pixel** [comment]
hii
